rs/bitcoin/adapter/src/connectionmanager.rs

```diff
@@ -662,11 +674,13 @@ impl<Block: Clone> ProcessEvent for ConnectionManager<NetworkMessage<Block>> {
     }
 }
 
-impl<Block: Clone> ProcessNetworkMessage<Block> for ConnectionManager<NetworkMessage<Block>> {
+impl<Network: BlockchainNetwork> ProcessNetworkMessage<Network::Header, Network::Block>
```

Comment:
In cases where there is a struct, that is generic over `<Network::Header, Network::Block>`, we might as well make it generic over `<Network>`, and then use the associated types internally. I don't see a usecase, where there would be a `<Bitcoin::Header, Dogecoin::Block>`. 

So this might make the code more readable and robust.

Reply:
Yes, in this case it can be simplified. Fixed